src/Illuminate/Console/Signals.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace Illuminate\Console;
+
+use Symfony\Component\Console\SignalRegistry\SignalRegistry;
+
+class Signals
+{
+    /**
+     * The signal registry instance.
+     *
+     * @var \Symfony\Component\Console\SignalRegistry\SignalRegistry
+     */
+    protected $registry;
+
+    /**
+     * The signal registry's previous list of handlers.
+     *
+     * @param array<int, array<int, callable>>|null
+     */
+    protected $previousHandlers;
+
+    /**
+     * Create a new Signals instance.
+     *
+     * @param  \Symfony\Component\Console\SignalRegistry\SignalRegistry  $registry
+     * @return void
+     */
+    public function __construct($registry)
```

Comment:
@nunomaduro current Laravel version requires PHP8+, so why don't you use strict types here (and many other places)? Why not:

```suggestion
    public function __construct(SignalRegistry $registry)
```

Reply:
No plans on adding a type-hint here. 👍🏻

Reply:
But why? I'm just curious, because there are multiple places in this PR where language-level types (arguments, properties, returns) could be added which would make this code just better (safer). Is it Laravel's convention 🤔?

Reply:
Backward compatibility i guess ? i dont know...

Reply:
In brand new files 😉? That's the point, I saw Nuno's tweet, looked and the changes out of curiosity and saw this. It does not make sense to me, so I asked to understand 🤷